src/harness/harness.c

```diff
@@ -34,25 +34,141 @@ tHarness_game_info harness_game_info;
+    76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 172, 95, 43, 8, 13, 13, 0, 63, 124, 58, 64, 62, 44, 123, 125, 126, 27,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 47, 42, 45, 43, 46, 0, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 32, 8,
+};
+
 int Harness_ProcessCommandLine(int* argc, char* argv[]);
 
 void Harness_DetectGameMode() {
-    if (access("DATA/RACES/CITY01.TXT", F_OK) == -1 && access("DATA/RACES/CITYA1.TXT", F_OK) == -1) {
-        harness_game_info.defines.INTRO_SMK_FILE = "";
-        harness_game_info.defines.GERMAN_LOADSCRN = "COWLESS.PIX";
-        harness_game_info.mode = eGame_carmageddon_demo;
-        LOG_INFO("\"%s\"", "Carmageddon demo");
-    } else if (access("DATA/CUTSCENE/SPLINTRO.SMK", F_OK) != -1) {
-        harness_game_info.defines.INTRO_SMK_FILE = "SPLINTRO.SMK";
-        harness_game_info.defines.GERMAN_LOADSCRN = "LOADSCRN.PIX";
-        harness_game_info.mode = eGame_splatpack;
-        LOG_INFO("\"%s\"", "Splat Pack");
+    if (access("DATA/RACES/CASTLE.TXT", F_OK) != -1) {
+        // All splatpack edition have the castle track
+        if (access("DATA/RACES/CASTLE2.TXT", F_OK) != -1) {
+            // Only the full splat release has the castle2 track
+            harness_game_info.defines.INTRO_SMK_FILE = "SPLINTRO.SMK";
+            harness_game_info.defines.GERMAN_LOADSCRN = "LOADSCRN.PIX";
+            harness_game_info.mode = eGame_splatpack;
+            LOG_INFO("\"%s\"", "Splat Pack");
+        } else if (access("DATA/RACES/TINSEL.TXT", F_OK) != -1) {
+            // Only the the splat x-mas demo has the tinsel track
+            harness_game_info.defines.INTRO_SMK_FILE = "";
+            harness_game_info.defines.GERMAN_LOADSCRN = "";
+            harness_game_info.mode = eGame_splatpack_demo;
+            LOG_INFO("\"%s\"", "Splat Pack X-mas demo");
+        } else {
+            // Assume we're using the splatpack demo
+            harness_game_info.defines.INTRO_SMK_FILE = "";
+            harness_game_info.defines.GERMAN_LOADSCRN = "";
+            harness_game_info.mode = eGame_splatpack_demo;
+            LOG_INFO("\"%s\"", "Splat Pack demo");
+        }
+    } else if (access("DATA/RACES/CITYB3.TXT", F_OK) != -1) {
+        // All non-splatpack edition have the cityb3 track
+        if (access("DATA/RACES/CITYA1.TXT", F_OK) == -1) {
+            // The demo does not have the citya1 track
+            harness_game_info.defines.INTRO_SMK_FILE = "";
+            harness_game_info.defines.GERMAN_LOADSCRN = "COWLESS.PIX";
+            harness_game_info.mode = eGame_carmageddon_demo;
+            LOG_INFO("\"%s\"", "Carmageddon demo");
+        }
+        else {
+            goto carmageddon;
+        }
     } else {
-        harness_game_info.defines.INTRO_SMK_FILE = "MIX_INTR.SMK";
+carmageddon:
+        if (access("DATA/CURSCENE/Mix_intr.smk", F_OK) == -1) {
```

Comment:
`CURSCENE` should be `CUTSCENE`?

Also, do we need this now we've fixed the case sensitive file access?

Reply:
Indeed, can be collapsed into one.

Reply:
And the `INTO_SMK_FILE` variable can be set unconditionally

Reply:
And the `INTO_SMK_FILE` variable can be set unconditionally

Reply:
I need to patch `smk_open_file` for that.
It looks like that file has already been customized.